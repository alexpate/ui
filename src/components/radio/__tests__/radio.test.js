/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';
import {shallow} from 'enzyme';

import theme from 'components/theme';
import Radio from 'components/radio';
import Text from 'components/text';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Radio', () => {
  const defaultProps = {
    size: 'medium',
    checked: true,
  };
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Radio {...defaultProps} {...props} />
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should allow a custom label to be passed in', () => {
    const wrapper = shallow(
      <Radio
        label={
          <Text size="extra-large" color="green500">
            A custom label
          </Text>
        }
      />
    );

    expect(
      wrapper
        .find('Text')
        .childAt(0)
        .text()
    ).toBe('A custom label');
  });
});
