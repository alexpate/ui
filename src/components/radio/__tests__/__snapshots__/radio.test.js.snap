// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render shallow component ok 1`] = `
.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="ui-radio ui-radio--medium emotion-0 emotion-1"
  data-test="ui-radio"
  name={undefined}
  style={Object {}}
>
  <input
    checked={true}
    disabled={undefined}
    id="radio1"
    onChange={[Function]}
    type="radio"
  />
  <label
    htmlFor="radio1"
    onClick={undefined}
  />
</div>
`;
